litestar/response/sse.py

```diff
@@ -14,21 +16,26 @@
     from litestar.types import ResponseCookies, ResponseHeaders, StreamType
 
 _LINE_BREAK_RE = re.compile(r"\r\n|\r|\n")
+DEFAULT_SEPARATOR = "\r\n"
 
 
 class _ServerSentEventIterator(AsyncIteratorWrapper[bytes]):
-    __slots__ = ("content_async_iterator",)
+    __slots__ = ("content_async_iterator", "event_id", "event_type", "retry_duration", "comment_message")
 
     content_async_iterator: AsyncIteratorWrapper[bytes | str] | AsyncIterable[str | bytes] | AsyncIterator[str | bytes]
 
     def __init__(
         self,
         content: str | bytes | StreamType[str | bytes],
         event_type: str | None = None,
-        event_id: int | None = None,
+        event_id: str | None = None,
```

Comment:
This will have to be

```suggestion
        event_id: int | str | None = None,
```

at least for the time being. You can keep it as a `str` internally, but the public API needs to be backwards compatible.